cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -29,7 +29,7 @@ type Watcher struct {
 }
 
 func New(cfg *config.Instance, an *sync.Client) *Watcher {
-	w := &Watcher{an: an, stop: make(chan struct{}), cfg: cfg, interval: defaultInterval}
+	w := &Watcher{an: an, stop: make(chan struct{}, 1), cfg: cfg, interval: defaultInterval}
```

Comment:
nit: Is this buffer on the channel necessary? We could potentially remove the buffer and simply close the channel in the `Close()` function. Same for the `stop` channel in the `DotProgress`

Reply:
It's effectively a failsafe. I don't think it's necessary with how we're using this channel, but it's better to be explicit.